Yaegi sometimes treats variables of a named array type as structs

I'm working on [gitlab.com/eukano/runway](https://gitlab.com/eukano/runway), in which Yaegi serves as an interpreter for scripts that draw 3D models. The interpreter is set up [here](https://gitlab.com/eukano/runway/-/blob/568c19d470a5694769ec17c9bba7b570b2331b2b/script/interpreter/interpreter.go). An example model/script is [here](https://gitlab.com/eukano/runway/-/blob/568c19d470a5694769ec17c9bba7b570b2331b2b/example/simple.runway).

1. In the interpreter setup, I evaluate `type Vec = geom.Vec`
2. `geom.Vec` is defined as `type Vec = mgl32.Vec`
3. `mgl32.Vec` is defined as `type Vec3 f32.Vec3`
4. `[bug]` In the script, calling a method on a `Vec` causes a panic:

```
reflect: call of reflect.Value.FieldByIndex on array Value
goroutine 1 [running, locked to thread]:
reflect.flag.mustBe(...)
        /usr/lib/go/src/reflect/value.go:221
reflect.Value.FieldByIndex(0x1397ca0, 0xc0006f2e50, 0x191, 0xc0006f2838, 0x0, 0x0, 0xc0006eb600, 0xc0006e8b90, 0x1397ca0)
        /usr/lib/go/src/reflect/value.go:884 +0x305
github.com/traefik/yaegi/interp.getIndexSeqMethod.func2(0xc0002a6210, 0xc0006e8b90)
        REDACTED/src/yaegi/interp/run.go:1996 +0xd8
```

For example, `Vec{1,2,3}.Normalize()` or `var v Vec; v.Normalize()` cause this error. This can be reproduced by cloning Runway and running `go run ./cmd/runway inspect <script>` where the script contains a Vec method call. Compiling/running with `-tags nogtk,noegl` should avoid the CGo depedencies.

Tracing this backwards:

1. The error is raised by `FieldByIndex` because `value(f)` is not a struct

    https://github.com/traefik/yaegi/blob/2a0d29a390dbfa728eb01376d70e62ff9aacb1fd/interp/run.go#L1995-L1998

2. That function is assigned to `n.exec` because the method receiver is not a pointer

    https://github.com/traefik/yaegi/blob/2a0d29a390dbfa728eb01376d70e62ff9aacb1fd/interp/run.go#L1989

3. `getIndexSeqMethod` is assigned to `n.gen` as part of control flow graph generation

   https://github.com/traefik/yaegi/blob/2a0d29a390dbfa728eb01376d70e62ff9aacb1fd/interp/cfg.go#L1589-L1598

I reproduced this with a test:

```go

type Issue1149_Vec3 [3]float32

func (v Issue1149_Vec3) LenSqr() float32 { return v[0]*v[0] + v[1]*v[1] + v[2]*v[2] }

func TestIssue1149(t *testing.T) {
	i := interp.New(interp.Options{})
	i.Use(interp.Exports{
		"geom/geom": map[string]reflect.Value{
			"Vec": reflect.ValueOf((*Issue1149_Vec3)(nil)),
		},
	})
	i.ImportUsed()

	_, err := i.Eval(`
		type Vec = geom.Vec
	`)
	if err != nil {
		t.Fatal(err)
	}

	runTests(t, i, []testCase{
		{src: "Vec{1, 2, 3}.LenSqr()"},
	})
}
```

```console
$ go test -run=TestIssue1149 ./interp/
1:28: panic
--- FAIL: TestIssue1149 (0.00s)
    --- FAIL: TestIssue1149/#00 (0.00s)
        interp_eval_test.go:784: got an error: reflect: call of reflect.Value.FieldByIndex on array Value
        interp_eval_test.go:784: goroutine 6 [running]:
            runtime/debug.Stack(0x1, 0xc0002fe600, 0x40)
                /usr/lib/go/src/runtime/debug/stack.go:24 +0x9f
            github.com/traefik/yaegi/interp.(*Interpreter).eval.func1(0xc00012ddd8)
                /home/firelizzard/src/yaegi/interp/interp.go:503 +0xc8
            panic(0xdd99a0, 0xc00000e420)
                /usr/lib/go/src/runtime/panic.go:965 +0x1b9
            github.com/traefik/yaegi/interp.runCfg.func1(0xc0003460b0, 0xc0000c1400, 0xc00012dbb8)
                /home/firelizzard/src/yaegi/interp/run.go:185 +0x253
            panic(0xdd99a0, 0xc00000e420)
                /usr/lib/go/src/runtime/panic.go:965 +0x1b9
            reflect.flag.mustBe(...)
                /usr/lib/go/src/reflect/value.go:221
            reflect.Value.FieldByIndex(0xdfa340, 0xc000036c24, 0x191, 0xc000036ba0, 0x0, 0x0, 0xc000036be0, 0xc0000a8800, 0xc0000322d0)
                /usr/lib/go/src/reflect/value.go:884 +0x305
            github.com/traefik/yaegi/interp.getIndexSeqMethod.func2(0xc0003460b0, 0xc0000322d0)
                /home/firelizzard/src/yaegi/interp/run.go:2002 +0xa5
            github.com/traefik/yaegi/interp.runCfg(0xc0000c1400, 0xc0003460b0)
                /home/firelizzard/src/yaegi/interp/run.go:191 +0x87
            github.com/traefik/yaegi/interp.(*Interpreter).run(0xc00000c1e0, 0xc0000c1000, 0x0)
                /home/firelizzard/src/yaegi/interp/run.go:122 +0x2b0
            github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc00000c1e0, 0xeee843, 0x15, 0x0, 0x0, 0x7f47d858a101, 0x0, 0x0, 0x0, 0x0, ...)
                /home/firelizzard/src/yaegi/interp/interp.go:585 +0x6fe
            github.com/traefik/yaegi/interp.(*Interpreter).Eval(...)
                /home/firelizzard/src/yaegi/interp/interp.go:401
            github.com/traefik/yaegi/interp_test.assertEval(0xc000001b00, 0xc00000c1e0, 0xeee843, 0x15, 0x0, 0x0, 0x0, 0x0)
                /home/firelizzard/src/yaegi/interp/interp_eval_test.go:806 +0x92
            github.com/traefik/yaegi/interp_test.runTests.func1(0xc000001b00)
                /home/firelizzard/src/yaegi/interp/interp_eval_test.go:784 +0x9d
            testing.tRunner(0xc000001b00, 0xc00000e300)
                /usr/lib/go/src/testing/testing.go:1193 +0xef
            created by testing.(*T).Run
                /usr/lib/go/src/testing/testing.go:1238 +0x2b3
FAIL
FAIL    github.com/traefik/yaegi/interp 0.004s
FAIL
```

--- a/interp/interp_eval_test.go
+++ b/interp/interp_eval_test.go
@@ -1511,3 +1511,30 @@ func TestIssue1142(t *testing.T) {
 		{src: "a := 1; // foo bar", res: "1"},
 	})
 }
+
+type Issue1149_Array [3]float32
+
+func (v Issue1149_Array) Foo() string  { return "foo" }
+func (v *Issue1149_Array) Bar() string { return "foo" }
+
+func TestIssue1149(t *testing.T) {
+	i := interp.New(interp.Options{})
+	i.Use(interp.Exports{
+		"pkg/pkg": map[string]reflect.Value{
+			"Type": reflect.ValueOf((*Issue1149_Array)(nil)),
+		},
+	})
+	i.ImportUsed()
+
+	_, err := i.Eval(`
+		type Type = pkg.Type
+	`)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	runTests(t, i, []testCase{
+		{src: "Type{1, 2, 3}.Foo()", res: "foo"},
+		{src: "Type{1, 2, 3}.Bar()", res: "foo"},
+	})
+}

--- a/interp/interp_eval_test.go
+++ b/interp/interp_eval_test.go
@@ -1512,16 +1512,16 @@ func TestIssue1142(t *testing.T) {
 	})
 }
 
-type Issue1149_Array [3]float32
+type Issue1149Array [3]float32
 
-func (v Issue1149_Array) Foo() string  { return "foo" }
-func (v *Issue1149_Array) Bar() string { return "foo" }
+func (v Issue1149Array) Foo() string  { return "foo" }
+func (v *Issue1149Array) Bar() string { return "foo" }
 
 func TestIssue1149(t *testing.T) {
 	i := interp.New(interp.Options{})
 	i.Use(interp.Exports{
 		"pkg/pkg": map[string]reflect.Value{
-			"Type": reflect.ValueOf((*Issue1149_Array)(nil)),
+			"Type": reflect.ValueOf((*Issue1149Array)(nil)),
 		},
 	})
 	i.ImportUsed()

--- a/interp/run.go
+++ b/interp/run.go
@@ -1965,14 +1965,28 @@ func getIndexSeqPtrMethod(n *node) {
 	next := getExec(n.tnext)
 
 	if n.child[0].typ.TypeOf().Kind() == reflect.Ptr {
-		n.exec = func(f *frame) bltn {
-			getFrame(f, l).data[i] = value(f).Elem().FieldByIndex(fi).Addr().Method(mi)
-			return next
+		if len(fi) == 0 {
+			n.exec = func(f *frame) bltn {
+				getFrame(f, l).data[i] = value(f).Method(mi)
+				return next
+			}
+		} else {
+			n.exec = func(f *frame) bltn {
+				getFrame(f, l).data[i] = value(f).Elem().FieldByIndex(fi).Addr().Method(mi)
+				return next
+			}
 		}
 	} else {
-		n.exec = func(f *frame) bltn {
-			getFrame(f, l).data[i] = value(f).FieldByIndex(fi).Addr().Method(mi)
-			return next
+		if len(fi) == 0 {
+			n.exec = func(f *frame) bltn {
+				getFrame(f, l).data[i] = value(f).Addr().Method(mi)
+				return next
+			}
+		} else {
+			n.exec = func(f *frame) bltn {
+				getFrame(f, l).data[i] = value(f).FieldByIndex(fi).Addr().Method(mi)
+				return next
+			}
 		}
 	}
 }
@@ -1987,14 +2001,28 @@ func getIndexSeqMethod(n *node) {
 	next := getExec(n.tnext)
 
 	if n.child[0].typ.TypeOf().Kind() == reflect.Ptr {
-		n.exec = func(f *frame) bltn {
-			getFrame(f, l).data[i] = value(f).Elem().FieldByIndex(fi).Method(mi)
-			return next
+		if len(fi) == 0 {
+			n.exec = func(f *frame) bltn {
+				getFrame(f, l).data[i] = value(f).Elem().Method(mi)
+				return next
+			}
+		} else {
+			n.exec = func(f *frame) bltn {
+				getFrame(f, l).data[i] = value(f).Elem().FieldByIndex(fi).Method(mi)
+				return next
+			}
 		}
 	} else {
-		n.exec = func(f *frame) bltn {
-			getFrame(f, l).data[i] = value(f).FieldByIndex(fi).Method(mi)
-			return next
+		if len(fi) == 0 {
+			n.exec = func(f *frame) bltn {
+				getFrame(f, l).data[i] = value(f).Method(mi)
+				return next
+			}
+		} else {
+			n.exec = func(f *frame) bltn {
+				getFrame(f, l).data[i] = value(f).FieldByIndex(fi).Method(mi)
+				return next
+			}
 		}
 	}
 }

